app/Console/Commands/RegisterFamiliesFromFile.php

```diff
@@ -0,0 +1,196 @@
+    protected $description = 'Imports and registers families against a CC from a CSV against a Children\'s Centre';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        // Load the CSV
+        $this->rows = Excel::load($this->argument('file'))->get();
+
+        // Search for existing user and centre
+        $this->user = User::where('email', $this->argument('email'))->first();
+        $this->centre = Centre::where('prefix', $this->argument('prefix'))->first();
+
+        switch (true) {
+            case (!$this->user):
+                exit("Can't find that User.\n");
+                break;
+            case (!$this->centre):
+                exit("Can't find the Centre.\n");
+                break;
+            case ($this->rows->isEmpty()):
+                exit("There are no registrations.\n");
+                break;
+            default:
+                // Check the user is happy to proceed
+                if (!$this->warnUser()) {
+                    exit("Exit without change.\n");
+                };
+
+                // Log the user in.
+                Auth::login($this->user);
+                if (!Auth::check()) {
+                    exit("Failed to login.\n");
+                };
+
+                // Is it one of the User's Centres?
+                if (!Auth::user()->isRelevantCentre($this->centre)) {
+                    exit("User cannot access that Centre.\n");
+                };
+
+                $bar = $this->output->createProgressBar($this->rows->count());
+
+                $this->rows->each(function (CellCollection $row) use ($bar) {
+
+                    $signup_date = Carbon::createFromFormat('Y-m-d', $row->signup_date)->startOfDay();
+
+                    $registration = new Registration([
+                        'fm_chart_on' => ($row->pie_chart == "Y") ? $signup_date : null,
+                        'fm_diary_on' => ($row->pie_chart == "Y") ? $signup_date : null,
+                        'consented_on' => $signup_date,
+                        'eligibility' => 'other',
+                        ''
+                    ]);
+
+                    // Make a new family.
+                    $family = new Family();
+
+                    $family->initial_centre_id = $this->centre->id;
+
+                    // Cut the prefix and cast left-padded 0's to an int.
+                    $family->centre_sequence = (int) preg_replace(
+                        "/". $this->centre->prefix . "/",
+                        "",
+                        $row->seq_reference
+                    );
+
+                    // Make some carers
+                    $carer = $row->primary_carer_name;
+                    $carers = array_values(
+                        array_filter(
```

Comment:
Hard to follow by just reading code without seeing data too... assuming you have tested.

Reply:
I've run it through the real stuff they've said we can see. *seems* to work...

Reply:
Awesome